### PR TITLE
🧪 Add tests for _is_safe_ip in feed_service

### PR DESCRIPTION
**Pull request by sheepdestroyer**
🎯 **What:** Adds missing test coverage for the `_is_safe_ip` function in `backend/feed_service.py` to ensure proper behavior in identifying safe/unsafe IPv4 and IPv6 addresses.
📊 **Coverage:** Tests various scenarios including happy paths (public IPs) and edge cases/error paths (private, loopback, link-local, multicast, etc.) using pytest parametrization.
✨ **Result:** Improved test coverage and regression prevention for critical IP validation functionality.

---
*PR created automatically by Jules for task [16934706835428533713](https://jules.google.com/task/16934706835428533713) started by @sheepdestroyer*

## Summary by Sourcery

Tests:
- Introduce parametrized tests verifying _is_safe_ip behavior for various IPv4 and IPv6 public, private, loopback, link-local, multicast, reserved, and unspecified addresses.

## Release Notes

* **Chores**
  * Updated agent-tools subproject dependency

* **Tests**
  * Added comprehensive unit tests for IP address validation

[Review Change Stack](https://app.coderabbit.ai/change-stack/sheepdestroyer/SheepVibes/pull/470?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: agent-tools [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit e4cefe4bf5740bd6f4b726786d3848444e2e9c4e
+Subproject commit a8d240042ced107b338d5987414725d59b877918
```
Comment (sheepdestroyer) [inline agent-tools:1]: This file bumps the agent-tools submodule pointer from `e4cefe4` to `a8d2400`. This change is unrelated to the `_is_safe_ip` test additions and should be reverted from this branch. Submodule changes belong in their own focused PR with context explaining why the update is needed.

**Review by sheepdestroyer — COMMENTED** [review]
## Code Review Summary

**Verdict: Changes Suggested** (1 issue to fix, 3 suggestions)

### Issue to fix before merge

- **agent-tools** — This PR bumps the agent-tools submodule pointer (e4cefe4 -> a8d2400) which is completely unrelated to the _is_safe_ip tests. Submodule updates belong in their own focused PR. Please revert this change from the branch.

### Suggestions (non-blocking)

- **tests/unit/test_safe_ip.py** — Consider adding CGNAT (100.64.0.0/10, e.g. `100.64.0.1`) and broadcast (`255.255.255.255`) test cases to fully cover SSRF protections. Gemini Code Assist also flagged IPv4-mapped IPv6 addresses (`::ffff:10.0.0.1`) as worth testing, though this is more of a hardening measure.
- **tests/unit/test_safe_ip.py** — Add `ids=` to `@pytest.mark.parametrize` so each test case has a descriptive name in pytest output. This makes CI failures easier to diagnose (Sourcery also flagged this).
- **tests/unit/test_safe_ip.py:25** — Add `# noqa: S104` to the `("0.0.0.0", False)` line to suppress the Ruff false positive about binding to 0.0.0.0 (this is test data, not an actual bind).

### Looks Good
- Clean, well-structured test with comprehensive category coverage (12 IPv4 + 8 IPv6 cases)
- All 20 tests pass; CI is green across all checks (Backend Tests, Codacy, DeepSource, Sourcery, CodeRabbit)
- Good parametrization pattern and correct imports from the module under test
- The `_is_safe_ip` implementation correctly checks all 6 `is_*` properties on the ipaddress object
